jobs/contact-retrieve-transcript/index.ts

```diff
@@ -83,7 +84,7 @@ export const processRecordWithoutException = async (sqsRecord: SQSRecord): Promi
   try {
     await processRecord(message);
   } catch (err) {
-    console.error('Failed to process record', err);
+    console.error(new ContactJobProcessorError('Failed to process record'), err);
```

Comment:
I'm assuming that setting specific levels is out of scope for this PR and we are just refactoring to use consistent error types for now? I'm just a little concerned that we are using 'error' level a little too liberally, especially in the case of job failures

There are some cases where jobs failing might not be unexpected. For example, transcript imports might have to wait for other async, eventually consistent systems to get their ducks in a row before the job can be successful (and these being external, Twilio ducks means we can't easily establish their state ahead of time), so the job might need to be tried a couple of times.

I know the best practice would be to handle the 'expected' failure states in the job logic, returning a fail message and not let it get to this catch block. Even so, I wonder if an error for a single attempt of a job failing, even for an unexpected reason, should be an error - or whether that should be a warning and only be an error when we give up retrying

Not for this PR though. We should do a separate sweep of our system to ensure we are logging at the standard levels and to fill in gaps in our logging coverage

Reply:
Yes! I've been tracking these small-ish thing that still need to be done (or nits I run into, but don't feel like fit into a PR) related to transcripts, but aren't really ready for a full ticket in https://benetech.app.box.com/file/1068884092948. I've added this task to that list.